5.analyze-weights/gsea.py

```diff
@@ -0,0 +1,117 @@
+"""
+2018 Gregory Way
+5.analyze-components/gsea.py
+
+Submit jobs that perform GSEA on all compiled weight matrices
+
+Usage:
+
+    Run in command line:
+
+            python gsea.py
+
+     with optional arguments:
+       --pmacs_config       filepath pointing to PMACS configuration file
+       --python_path        absolute path of PMACS python in select environment
+                              default: '~/.conda/envs/tybalt-gpu/bin/python'
+       --local              if provided, sweep will be run locally instead
+
+Output:
+
+    Will submit jobs for each ensemble of compressed weight matrix (directory)
+"""
+
+import os
+import sys
+import argparse
+import pandas as pd
+
+sys.path.insert(0, '../scripts/util')
+from bsub_helper import bsub_help
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-c', '--pmacs_config',
+                    default='../config/pmacs_config.tsv',
+                    help='location of the configuration file for PMACS')
+parser.add_argument('-p', '--python_path', help='absolute path of python',
+                    default='python')
+parser.add_argument('-l', '--local', action='store_true',
+                    help='decision to run models locally instead of on PMACS')
+args = parser.parse_args()
+
+pmacs_config_file = args.pmacs_config
+python_path = args.python_path
+local = args.local
+
+# Load data
+config_df = pd.read_table(pmacs_config_file, index_col=0)
+
+# Retrieve PMACS configuration
+queue = config_df.loc['queue']['assign']
+num_gpus = config_df.loc['num_gpus']['assign']
```

Comment:
does this implementation of gsea work with gpus?

Reply:
not optimized to, but yes, will run